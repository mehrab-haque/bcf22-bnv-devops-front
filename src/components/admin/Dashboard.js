import * as React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';
import Divider from '@mui/material/Divider';
import Drawer from '@mui/material/Drawer';
import IconButton from '@mui/material/IconButton';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import MenuIcon from '@mui/icons-material/Menu';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import {useState} from "react";
import ForumIcon from '@mui/icons-material/Forum';
import AddAlertIcon from '@mui/icons-material/AddAlert';
import LocationCityIcon from '@mui/icons-material/LocationCity';
import DeliveryDiningIcon from '@mui/icons-material/DeliveryDining';
import AssessmentIcon from '@mui/icons-material/Assessment';
import PeopleAltIcon from '@mui/icons-material/PeopleAlt';
import ExitToAppIcon from '@mui/icons-material/ExitToApp';
import {Button, CircularProgress, Dialog, DialogActions, DialogContent, DialogTitle} from "@mui/material";
import Courses from "./Courses";

const drawerWidth = 240;

var showCart

const Dashboard=props=>{
    const { window } = props;
    const [mobileOpen, setMobileOpen] = React.useState(false);

    const handleDrawerToggle = () => {
        setMobileOpen(!mobileOpen);
    };

    const [menu,setMenu]=useState(0)

    const drawer = (
        <div>
            <Toolbar />

            <List>
                <Divider />
                <ListItem button selected={menu===1} onClick={()=>{setMenu(1)}}>
                    <ListItemIcon>
                        <ForumIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Courses'} />
                </ListItem>
                <ListItem button selected={menu===2} onClick={()=>{setMenu(2)}}>
                    <ListItemIcon>
                        <LocationCityIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Students'} />
                </ListItem>
                <Divider />
                <ListItem button selected={menu===3} onClick={()=>{setMenu(3)}}>
                    <ListItemIcon>
                        <AddAlertIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Courses Assignment'} />
                </ListItem>
                <Divider />
                <ListItem button selected={menu===4} onClick={()=>{setMenu(4)}}>
                    <ListItemIcon>
                        <DeliveryDiningIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Result'} />
                </ListItem>
                <Divider />
            </List>
        </div>
    );

    const container = window !== undefined ? () => window().document.body : undefined;


    return (
        <Box sx={{ display: 'flex' }}>
            <CssBaseline />
            <AppBar
                position="fixed"
                sx={{
                    width: { sm: `calc(100% - ${drawerWidth}px)` },
                    ml: { sm: `${drawerWidth}px` },
                }}
                style={{backgroundColor:'#666666'}}
            >
                <Toolbar>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        edge="start"
                        onClick={handleDrawerToggle}
                        sx={{ mr: 2, display: { sm: 'none' } }}
                    >
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" noWrap component="div">
                        Admin Dashboard
                    </Typography>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        edge="end"
                        sx={{marginLeft:'auto'}}
                        onClick={props.logout}
                    >
                        <ExitToAppIcon/>
                    </IconButton>
                </Toolbar>
            </AppBar>
            <Box
                component="nav"
                sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
                aria-label="mailbox folders"
            >
                {/* The implementation can be swapped with js to avoid SEO duplication of links. */}
                <Drawer
                    container={container}
                    variant="temporary"
                    open={mobileOpen}
                    onClose={handleDrawerToggle}
                    ModalProps={{
                        keepMounted: true, // Better open performance on mobile.
                    }}
                    sx={{
                        display: { xs: 'block', sm: 'none' },
                        '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth },
                    }}
                >
                    {drawer}
                </Drawer>
                <Drawer
                    variant="permanent"
                    sx={{
                        display: { xs: 'none', sm: 'block' },
                        '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth },
                    }}
                    open
                >
                    {drawer}
                </Drawer>
            </Box>
            <Box
                component="main"
                sx={{ flexGrow: 1, p: 3, width: { sm: `calc(100% - ${drawerWidth}px)` } }}
            >
                <Toolbar />


                {
                    menu===1?(
                        <Courses/>
                    ):(
                        <div/>
                    )
                }




            </Box>
        </Box>
    );
}

Dashboard.propTypes = {
    /**
     * Injected by the documentation to work in an iframe.
     * You won't need it on your project.
     */
    window: PropTypes.func,
};

export default Dashboard
export {showCart}