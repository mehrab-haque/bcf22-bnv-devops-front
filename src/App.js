import "react-toastify/dist/ReactToastify.css";
import { ToastContainer, toast } from "react-toastify";
import {
  Routes,
  Route, useNavigate
} from "react-router-dom";
import Main from "./components/admin/Main";

var showToast;

function App() {



  showToast = message => {
    toast.dark(message, {
      position: "bottom-right",
      autoClose: 3000,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
      progress: undefined,
    });
  }

  return (
    <div>
      <ToastContainer
          position="bottom-right"
          autoClose={3000}
          hideProgressBar={false}
          newestOnTop={false}
          closeOnClick
          rtl={false}
          pauseOnFocusLoss
          draggable
          pauseOnHover
      />

      <Routes>
        <Route path="/admin" exact element={<Main />} />
      </Routes>

    </div>
  );
}

export default App;
export {showToast}
