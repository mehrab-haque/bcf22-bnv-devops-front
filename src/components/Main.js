import react, {useEffect} from 'react'
import {showToast} from "../App";

const Main=()=>{

    useEffect(()=>{
        showToast("Hello World")
        console.log("hi")
    },[])

    return(
        <div>
            Hello World
        </div>
    )
}

export default Main