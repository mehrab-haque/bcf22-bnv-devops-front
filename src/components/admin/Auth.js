import React, {useRef, useState} from 'react'
import {
    Button,
    Dialog,
    DialogActions,
    DialogContent,
    DialogTitle,
    Grid,
    LinearProgress,
    TextField
} from "@mui/material";
import {showToast} from "../../App";

const Auth=props=>{

    const loginRef=useRef();
    const passRef=useRef();

    const [loading,setLoading]=useState(false)

    const loginClick=()=>{
        setLoading(true)
        setTimeout(()=>{
            setLoading(false)
            props.login()
        }, 2000);
    }


    return(
        <div>
            <Dialog open={true}>
                {
                    loading?(
                        <LinearProgress/>
                    ):(
                        <div/>
                    )
                }
                <DialogTitle>
                    Admin Login
                </DialogTitle>
                <DialogContent>
                    <Grid container spacing={1}>
                        <Grid item xs={12}>
                            <TextField
                                inputRef={loginRef}
                                fullWidth
                                variant={'outlined'}
                                label={'Login'}
                                />
                        </Grid>
                        <Grid item xs={12}>
                            <TextField
                                inputRef={passRef}
                                fullWidth
                                type={'password'}
                                variant={'outlined'}
                                label={'Password'}
                            />
                        </Grid>
                    </Grid>
                </DialogContent>
                <DialogActions>
                    <Button
                        color={'primary'}
                        onClick={loginClick}
                        disabled={loading}
                        >
                        Login
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    )
}

export default Auth