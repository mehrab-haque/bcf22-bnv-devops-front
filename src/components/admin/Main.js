import react, {useEffect, useState} from 'react'
import {showToast} from "../../App";
import Dashboard from "./Dashboard";
import Auth from "./Auth";

var getToken

const Main=()=>{

    const [isAuth,setAuth]=useState(false)
    const [token,setToken]=useState(null)

    getToken=()=>{
        return token
    }

    const login=()=>{
        setAuth(true)
    }

    const logout=()=>{
        setAuth(false)
    }

    return(
        <div>
            {
                isAuth?(
                    <Dashboard logout={logout}/>
                ):(
                    <Auth login={login} setToken={setToken}/>
                )
            }
        </div>
    )
}

export default Main
export {getToken}