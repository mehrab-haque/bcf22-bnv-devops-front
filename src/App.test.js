import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import App from "./App";
import Auth from "./components/admin/Auth";
import Main from "./components/admin/Main";
import Dashboard from "./components/admin/Dashboard";

describe("Component App Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<App/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component Main Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Main/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component Auth Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Auth/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component Dashboard Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Dashboard/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});