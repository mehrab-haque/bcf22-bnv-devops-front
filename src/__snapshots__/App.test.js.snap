// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<div>
  <ToastContainer
    autoClose={3000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="bottom-right"
    role="alert"
    rtl={false}
    theme="light"
    transition={[Function]}
  />
  <Routes>
    <Route
      element={<Main />}
      exact={true}
      path="/admin"
    />
  </Routes>
</div>
`;
